### ISSUE TITLE
[Enhancement] Auto-detect last verse of chapter

### ISSUE THREAD

**Issue by ParkerRobb** [AnthropologieBiblique/link-heading-range#1]
Thank you so much for the work you've put into this (and the [Obsidian Theological Studies](https://github.com/AnthropologieBiblique/ObsidianTheologicalStudies) demo repository)! I recently set up my own Biblekasten in Obsidian, but I've been mulling over how linking to multiple verses could be done better and waiting for the community to figure it out. This plugin is an immense help, even in "rudimentary" alpha form.

I don't know if this would be possible given how the end verse of the range is implemented in the link, but it would be nice to be able to type a large number and have the link automatically snap to the last verse of the chapter. [Bible Linker](https://github.com/kuchejak/obsidian-bible-linker-plugin), the plugin that most people studying the Bible in Obsidian seem to use, just added this feature.

**Comment by Limezy** [comment]
Hi and thanks for your suggestion !
I didn't know that feature from Bible Linker, and it seems like a good idea indeed.
I guess it wouldn't be so hard, basically jump in to the linked note, parse the last h6 (= the last verse), if less do nothing, if more replace by that last h6.
The only thing is that as of now, this plugin is only working at rendering stage and has nothing to do with the source (contrarily to the Bible Linker plugin, that is basically a templater made as a plugin, hence acting at the source level), so the question would be when shall I trigger the parsing and replace ? Probably at the moment there is a switch from source to preview ?
I could try !

**Comment by Limezy** [comment]
>  but I've been mulling over how linking to multiple verses could be done better and waiting for the community to figure it out

That being said, @ParkerRobb have you looked at the two templaters that are demoed inside my demo repo ? I think they are a nice way to quote verse, because you can actually read and/or search them while quoting

**Comment by ParkerRobb** [comment]
> That being said, @ParkerRobb have you looked at the two templaters that are demoed inside my demo repo ? I think they are a nice way to quote verse, because you can actually read and/or search them while quoting

Yes, I just went through the tour of your demo vault and saw those. Wow, thanks for putting all that together! I like how you've implemented the ability to read/search the verses in those templaters, so I may switch from using the Bible Linker plugin to using those.
